src/CTA.Rules.Models/Tokens/NodeToken.cs

```diff
@@ -52,7 +52,28 @@ public NodeToken()
         public List<ProjectLevelAction> ProjectLevelActions { get; set; }
         public List<ProjectLevelAction> ProjectFileActions { get; set; }
 
-        public NodeToken Clone() => (NodeToken)this.MemberwiseClone();
+        public NodeToken Clone()
+        {
+            NodeToken cloned = (NodeToken)this.MemberwiseClone();
+            cloned.TextChanges = cloned.TextChanges?.Select(textChange => textChange.Clone()).ToList();
+            cloned.TargetCPU = cloned.TargetCPU?.ToList();
+            cloned.AttributeActions = cloned.AttributeActions?.Select(action => action.Clone<AttributeAction>())?.ToList();
+            cloned.AttributeListActions = cloned.AttributeListActions?.Select(action => action.Clone<AttributeAction>())?.ToList();
```

Comment:
Should the second one be attributelist?

Reply:
`AttributeListActions`'s type is `List<AttributeAction>`, so `AttributeAction` should be correct here.